### PR TITLE
Apm 292699 add option to filter incoming logs

### PR DESCRIPTION
**Pull request by mswiatkowska**

### CODE REVIEW

File: logs_ingest/filtering.py
Comment (pgutkowski) [inline logs_ingest/filtering.py:42]: Why not simple prefix check for supported types of group? right now second part of key doesn't matter, `FILTER.TEST.MIN_LOG_LEVEL.X=4` is evaluated the same as `FILTER.RESOURCE_TYPE.MIN_LOG_LEVEL.X=4`, I'm not sure if we want such a undefined behavior. Additionally, if we would introduce additional valid group type (other than `RESOURCE_TYPE` and `RESOURCE_ID`), it couldn't contain a comma.

File: logs_ingest/filtering.py
Comment (pgutkowski) [inline logs_ingest/filtering.py:86]: Really not sure about that, but I thought we would match contains with parsed `content` - if we match full record, possibly we could have a lots of false positives with record metadata: Let's say my application prints out word `category` in log lines that I want to push to Dynatrace - with current implementation I'm unable to select them, as `CONTAINS_PATTERN` would pass all records, as they all have `category` metadata field.

**Review by pgutkowski — CHANGES_REQUESTED** [review]

File: logs_ingest/filtering.py
Comment (mswiatkowska) [reply logs_ingest/filtering.py:86]: You're right, I should use `content ` for CONTAINS_PATTERN filter, I'll change that.

File: deployment/dynatrace-azure-logs.sh
Comment (sq2gxo) [inline deployment/dynatrace-azure-logs.sh:353]: FILTER_CONFIG should be optional - with thic code it became required?

Comment (mswiatkowska) [reply deployment/dynatrace-azure-logs.sh]: Thanks for catching that - I made it optional.

File: logs_ingest/filtering.py
Comment (pgutkowski) [inline logs_ingest/filtering.py:48]: Maybe I'm missing something, but why not merge this `if` section with the one at `44`, just cut the prefix to `FILTER.GLOBAL`?

File: logs_ingest/filtering.py
Comment (pgutkowski) [inline logs_ingest/filtering.py:67]: you can replace casefold call with simple `min_log_level`. The same applies to other places literals are used, like in `44`, `48` and `72`

File: deployment/dynatrace-azure-forwarder.json
Comment (pgutkowski) [inline deployment/dynatrace-azure-forwarder.json:98]: I think conflict resolution went wrong, because all these `tags` lines shouldn't be removed

**Review by pgutkowski — CHANGES_REQUESTED** [review]

**Comment by Dawid-Kaszubski** [comment]
Side note: I feel that this change is enough significant to bump up the forwarder version.

File: logs_ingest/mapping.py
Comment (sq2gxo) [inline logs_ingest/mapping.py:34]: to keep this dict consistent with 'log_level_to_severity_dict' you can replace it with:
`severity_to_log_level_dict = {v: k for k, v in log_level_to_severity_dict.items()}`

File: logs_ingest/filtering.py
Comment (mswiatkowska) [reply logs_ingest/filtering.py]: I simplified this but I needed to make small fix in other place;)

**Review by sq2gxo — APPROVED** [review]

**Review by Dawid-Kaszubski — APPROVED** [review]

**Review by pgutkowski — APPROVED** [review]

**Pull request merged**
